devito/ir/clusters/algorithms.py

```diff
@@ -222,33 +224,27 @@ def bump_and_contract(targets, source, sink):
 
 
 def clusterize(exprs):
-    """Group a sequence of :class:`ir.Eq`s into one or more :class:`Cluster`s."""
-    # Group expressions based on data dependences
-    groups = group_expressions(exprs)
-
+    """
+    Group a sequence of :class:`ir.Eq`s into one or more :class:`Cluster`s.
+    """
     clusters = ClusterGroup()
-
-    # Coerce iteration direction of each expression in each group
-    for g in groups:
-        mapper = OrderedDict([(e, e.directions) for e in g])
-        flowmap = detect_flow_directions(g)
-        queue = list(g)
-        while queue:
-            k = queue.pop(0)
-            directions, _ = force_directions(flowmap, lambda i: mapper[k].get(i))
-            directions = {i: directions[i] for i in mapper[k]}
-            # Need update propagation ?
-            if directions != mapper[k]:
-                mapper[k] = directions
-                queue.extend([i for i in g if i not in queue])
-
-        # Wrap each tensor expression in a PartialCluster
-        for k, v in mapper.items():
-            if k.is_Tensor:
-                scalars = [i for i in g[:g.index(k)] if i.is_Scalar]
-                intervals, sub_iterators, _ = k.ispace.args
-                ispace = IterationSpace(intervals, sub_iterators, v)
-                clusters.append(PartialCluster(scalars + [k], ispace, k.dspace))
+    for group in group_expressions(exprs):
```

Comment:
note for reviewers: this is so much nicers and simpler to understand than it used to...
note for @tjb900 : this is were I was hoping to have gained some speed. TBC